test/integration/crud/misc_cursors.test.js

```diff
@@ -3687,87 +2914,82 @@ describe('Cursor', function () {
       const cursor = collection.find();
       this.defer(() => cursor.close());
 
-      const promise = cursor.forEach(() => {});
+      const promise = cursor.forEach(() => { });
       expect(promise).to.exist.and.to.be.an.instanceof(Promise);
       return promise;
     });
   });
 
-  it('should return false when exhausted and hasNext called more than once', function (done) {
+  it('should return false when exhausted and hasNext called more than once', async function () {
     const configuration = this.configuration;
     const client = configuration.newClient({ w: 1 }, { maxPoolSize: 1 });
 
-    client.connect((err, client) => {
-      expect(err).to.not.exist;
-      this.defer(() => client.close());
+    await client.connect();
 
-      const db = client.db(configuration.db);
-      db.createCollection('cursor_hasNext_test').then(() => {
-        const cursor = db.collection('cursor_hasNext_test').find();
-        this.defer(() => cursor.close());
-
-        cursor
-          .hasNext()
-          .then(val1 => {
-            expect(val1).to.equal(false);
-            return cursor.hasNext();
-          })
-          .then(val2 => {
-            expect(val2).to.equal(false);
-            done();
-          });
-      });
-    });
+    const db = client.db(configuration.db);
+    await db.createCollection('cursor_hasNext_test');
+    const cursor = db.collection('cursor_hasNext_test').find();
+
+    const val1 = await cursor.hasNext();
+    expect(val1).to.equal(false);
+    const val2 = await cursor.hasNext();
+    expect(val2).to.equal(false);
+
+    await cursor.close();
+    await client.close();
   });
 
-  const testTransformStream = (config, _done) => {
+  const testTransformStream = async config => {
```

Comment:
suggest removing all these tests - we're now using the builtin `ReadableStream.map` method.  testing a transform on the stream's method is testing Nodejs, not our code